transmart-core-db-tests/src/main/groovy/org/transmartproject/db/StudyTestData.groovy

```diff
@@ -36,11 +36,15 @@ class StudyTestData {
                 if(!candidates) {
                     assert DimensionImpl.isBuiltinDimension(it), "Unknown dimension name '$it', " +
                             "modifier dimensions as string are not supported in createStudy()"
-                    study.addToDimensionDescriptions(new DimensionDescription(name: it))
+                    // saving is necessary here so that this dimension will be visible to other studies that are
+                    // created within this same transaction. Without it we will have duplicate dimensions.
+                    def dim = new DimensionDescription(name: it).save()
```

Comment:
It does. The test code first creates a bunch of studies with their associated data, and then saves all of them at once. The `DimensionDescription.findAllByName(it)` does not find unsaved dimension descriptions, so each study gets its own set of dimension descriptions. When all the studies are saved this results in a uniqueness violation.